File: tkk-dart/bin/nomor_dua.dart
import 'dart:io';

String emailCheck(String email){
  //Asumsi : Poin b - harus ada tanda titik '.' tepat setelah tanda '@'
  var validator = RegExp(r'^[A-Za-z0-9+_.-]{1,20}@[.][A-Za-z0-9.-]*[\\.co\\.id|\\.id]$');

  if(validator.hasMatch(email)){
    return 'Valid';
  }
  else{
    return 'Invalid';
  }
}

void main(List<String> arguments) {
  print('Masukkan Email : ');
  var email = stdin.readLineSync().toString();

  print(emailCheck(email));
}

File: tkk-java/src/tesKompetensiTeknis/NomorDua.java
package tesKompetensiTeknis;

import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class NomorDua {
    public static String emailCheck(String email)
    {
        //Asumsi : Poin b - harus ada tanda titik '.' tepat setelah tanda '@'
        String validator = "^[A-Za-z0-9+_.-]{1,20}@[.][A-Za-z0-9.-]*[\\.co\\.id|\\.id]$";

        Pattern pattern = Pattern.compile(validator);

        Matcher matcher = pattern.matcher(email);

        int length = email.length();

        if(matcher.matches()){
            return "Valid";
        }
        else{
            return "Invalid";
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Masukkan Email : ");

        String email = scanner.nextLine();

        System.out.println(emailCheck(email));

    }
}

